### ISSUE TITLE
grafana fails on Pivotal PKS

### ISSUE THREAD

**Issue by paulczar** [bitnami/charts#2352]
**Which chart**:

grafana 1.5.0

**Describe the bug**

When grafana is launched on a PKS cluster (with no ipv6 support) the image-renderer plugin fails and the pod crashes.

```
t=2020-04-17T15:30:49+0000 lvl=info msg="Browser initiated" logger=plugins.backend pluginId=grafana-image-renderer dumpio=false verboseLogging=false chromeBin=/opt/bitnami/grafana/data/plugins/grafana-image-renderer/chrome-linux/chrome ignoresHttpsErrors=false args=[--no-sandbox]
t=2020-04-17T15:30:49+0000 lvl=eror msg="Stopped RenderingService" logger=server reason="Failed to start renderer plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."

```

**To Reproduce**

1. Deploy to a cluster with no ipv6 support

**Expected behavior**

It should run.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7+vmware.1", GitCommit:"7be6fd53a1c9885bf9f470673f1c14280aab6386", GitTreeState:"clean", BuildDate:"2020-02-13T23:24:07Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

It appears to be related to this issue - https://github.com/grafana/grafana-image-renderer/issues/48

I worked around it by rebuilding the image using this Dockerfile:

```
FROM bitnami/grafana:6.7.2-debian-10-r13

USER root

RUN rm -rf /opt/bitnami/grafana/default-plugins/grafana-image-renderer
RUN rm -rf /opt/bitnami/grafana/data/plugins/grafana-image-renderer

USER 1001
➜  g 
➜  g 
➜  g cat Dockerfile
FROM bitnami/grafana:6.7.2-debian-10-r13

USER root

RUN rm -rf /opt/bitnami/grafana/default-plugins/grafana-image-renderer
RUN rm -rf /opt/bitnami/grafana/data/plugins/grafana-image-renderer

USER 1001

```

The stable/grafana helm chart works fine and doesn't have this problem, I assume its a difference in how the bitnami image vs the upstream grafana image are built.

**Comment by juan131** [comment]
Hi @paulczar 

It seems it's a known issue and they reported a workaround:

- https://github.com/grafana/grafana-image-renderer#known-issue-having-ipv6-disabled

As you can see in the link below, that plugins is not int the list of plugins included in the image by default, see:

- https://github.com/bitnami/bitnami-docker-grafana/blob/master/6/debian-10/rootfs/grafana-plugins.sh
- https://github.com/grafana/grafana-image-renderer/blob/master/docs/remote_rendering_using_docker.md

It seems the plugin is installed by Grafana during the initialization due to the default configuration (`grafana.ini`). You can add your custom configuration as explained in the link below, so you can avoid the plugin to be installed:

- https://github.com/bitnami/bitnami-docker-grafana#configuration

**Comment by paulczar** [comment]
Hi @juan131 its absolutely provided by the bitnami image by default.  Here's proof:

```
$ docker run --rm -d --name grafana  docker.io/bitnami/grafana:6.7.2-debian-10-r13 && sleep 1 && docker exec -ti grafana ls /opt/bitnami/grafana/default-plugins | grep image
1264bec01f455c03218a78c20bf07c1d121a7fe0936aaac84a9901e4451ad13e
digrich-bubblechart-panel	 pierosavi-imageit-panel
grafana-image-renderer		 scadavis-synoptic-panel
```
the default-plugins dir is created by the bitnami script you linked above (https://github.com/bitnami/bitnami-docker-grafana/blob/master/6/debian-10/rootfs/grafana-plugins.sh#L54-L55) and contains only what bitnami installs into it.

As I also mentioned the upstream grafana chart works fine, and in fact the `grafana/grafana:latest` doesn't contain the `grafana-image-renderer` plugin.  This is definitely a bitnami image issue, even if its not obvious from the image build scripts.

As far as I can tell, its not possible to disable a plugin at startup, which means the only way I can run this is by deleting the path `/opt/bitnami/grafana/default-plugins/grafana-image-renderer` in the image as shown in my prior example Dockerfile.

As it stands the bitnami/grafana chart (and image it runs) is unusable on many installations of Kubernetes including PKS/TKGI as its common to not support ipv6.

**Comment by juan131** [comment]
Hi @paulczar 

You're 100% right and I could identify the issue. We just released a new revision of the image (**6.7.2-debian-10-r18**) that does not ship the plugin by default. However, it'll try to install it by default during the initialization unless you overwrite the `GF_INSTALL_PLUGINS` environment variable.

You can find detailed information in the links below:

- https://github.com/bitnami/bitnami-docker-grafana/#install-plugins-at-initialization
- https://github.com/bitnami/bitnami-docker-grafana/#notable-changes

You can test the image in your side using the command below:

```console
$ docker run --rm -it --entrypoint bash docker.io/bitnami/grafana:6.7.2-debian-10-r18
$ ls /opt/bitnami/grafana/default-plugins
agenty-flowcharting-panel     digiapulssi-organisations-panel  jdbranham-diagram-panel		neocat-cal-heatmap-panel
bessler-pictureit-panel       digrich-bubblechart-panel        larona-epict-panel		pierosavi-imageit-panel
briangann-datatable-panel     fatcloud-windrose-panel	       marcuscalidus-svg-panel		satellogic-3d-globe-panel
briangann-gauge-panel	      grafana-clock-panel	       michaeldmoore-annunciator-panel	savantly-heatmap-panel
btplc-peak-report-panel       grafana-piechart-panel	       michaeldmoore-multistat-panel	scadavis-synoptic-panel
btplc-status-dot-panel	      grafana-polystat-panel	       mtanda-histogram-panel		snuids-radar-panel
btplc-trend-box-panel	      grafana-worldmap-panel	       natel-discrete-panel		snuids-trafficlights-panel
digiapulssi-breadcrumb-panel  gretamosa-topology-panel	       natel-plotly-panel		vonage-status-panel
```

**Comment by paulczar** [comment]
Great!  I believe I can override it using `extraEnvVars` in the helm chart.
